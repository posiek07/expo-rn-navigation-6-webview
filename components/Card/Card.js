import React from 'react';
import {TouchableOpacity} from 'react-native';
import moment from 'moment-timezone';
import {useNavigation} from '@react-navigation/native';
import propTypes from 'prop-types';
import {
  StyledCard,
  StyledTextWrapper,
  StyledTitleText,
  StyledDescriptionText,
  StyledDateText,
  StyledCardImageWrapper,
  StyledCardImage,
} from './style';

const Card = ({item}) => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity
      onPress={() =>
        navigation.navigate('EventDetailsScreen', {
          id: item.id,
        })
      }>
      <StyledCard>
        <StyledCardImageWrapper>
          <StyledCardImage
            accessibilityLabel={'image: ' + item.title}
            source={{uri: item.logoUrl}}
            alt={item.logoUrl}
          />
        </StyledCardImageWrapper>
        <StyledTextWrapper>
          <StyledDateText numberOfLines={1}>
            {moment(item.startDate).format('LLLL z')}
          </StyledDateText>
          <StyledTitleText numberOfLines={2}>{item.title}</StyledTitleText>
          <StyledDescriptionText numberOfLines={1}>
            {item.descriptionText}
          </StyledDescriptionText>
        </StyledTextWrapper>
      </StyledCard>
    </TouchableOpacity>
  );
};

Card.propTypes = {
  item: propTypes.object,
};

export default Card;
