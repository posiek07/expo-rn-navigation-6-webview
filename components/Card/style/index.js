import styled from 'styled-components/native';

export const StyledCard = styled.View`
  background-color: white;
  width: 100%;
  display: flex;
  flex-direction: row;
  margin-bottom: 5%;
  padding: 2%;
`;

export const StyledTextWrapper = styled.View`
  display: flex;
  width: 70%;
  padding-left: 2%;
  justify-content: space-between;
`;

export const StyledTitleText = styled.Text`
  color: black;
  font-size: 20px;
  font-weight: bold;
`;

export const StyledDescriptionText = styled.Text`
  color: grey;
`;

export const StyledDateText = styled.Text`
  color: #e85d04;
  font-size: 11px;
  font-weight: 700;
  margin: 3px;
`;

export const StyledCardImageWrapper = styled.View`
  background-color: black;
  width: 100px;
  height: 100px;
`;

export const StyledCardImage = styled.Image`
  width: 100%;
  height: 100%;
`;
