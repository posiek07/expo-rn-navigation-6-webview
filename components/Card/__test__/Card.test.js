import Card from '../Card';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';
import data from '../../../test-data/data.json';
import {render} from 'react-native-testing-library';
import {exact} from 'prop-types';

it.skip('should render and match snapshot', () => {
  const event = data.events[0];
  const card = renderer.create(<Card item={event} />).toJSON();
  expect(card).toMatchSnapshot();
});

it.skip('should render with title and image', async () => {
  const event = data.events[0];
  const {getByText, findByLabelText} = render(<Card item={event} />);

  const title = getByText('Evolving Design Systems');

  expect(title).toBeTruthy();

  const image = await findByLabelText('image: ' + data.events[0].name.text);

  expect(image).toBeTruthy();
});
