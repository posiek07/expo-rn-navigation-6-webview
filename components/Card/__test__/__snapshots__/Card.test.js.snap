// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": "5%",
          "paddingBottom": "2%",
          "paddingLeft": "2%",
          "paddingRight": "2%",
          "paddingTop": "2%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "height": 100,
            "width": 100,
          },
        ]
      }
    >
      <Image
        accessibilityLabel="image: undefined"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "paddingLeft": "2%",
            "width": "70%",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#e85d04",
              "fontSize": 11,
              "fontWeight": "700",
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
            },
          ]
        }
      >
        Sunday, March 27, 2022 4:25 PM 
      </Text>
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      />
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "grey",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
